main.tf

```diff
@@ -0,0 +1,175 @@
+terraform {
+  required_version = "~> 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+    template = {
+      source  = "hashicorp/template"
+      version = "~> 2.2.0"
+    }
+  }
+}
+
+
+provider "aws" {
+  region = var.aws_region
+  alias  = "aws"
+  assume_role {
+    role_arn = var.aws_role_arn
+  }
+}
+
+locals {
+  prefix             = var.prefix
+  availability_zones = ["${var.aws_region}a", "${var.aws_region}b", "${var.aws_region}c"]
+}
+
+resource "random_string" "naming" {
+  special = false
+  upper   = false
+  length  = 6
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    condition {
+      test     = "StringEquals"
+      values   = [var.databricks_account_id]
+      variable = "sts:ExternalId"
+    }
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::414351767826:root"]
```

Comment:
Should not be hardcoded

Reply:
its hardcoded from databricks, this is a databrick arn